sdk/cosmos/azure-cosmos/src/test/java/com/azure/cosmos/rx/GroupByQueryTests.java

```diff
@@ -0,0 +1,170 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+package com.azure.cosmos.rx;
+
+import com.azure.cosmos.CosmosAsyncClient;
+import com.azure.cosmos.CosmosAsyncContainer;
+import com.azure.cosmos.CosmosClientBuilder;
+import com.azure.cosmos.implementation.CosmosItemProperties;
+import com.azure.cosmos.implementation.Document;
+import com.azure.cosmos.models.FeedResponse;
+import com.azure.cosmos.models.ModelBridgeInternal;
+import com.azure.cosmos.models.QueryRequestOptions;
+import com.azure.cosmos.rx.pojos.City;
+import com.azure.cosmos.rx.pojos.Person;
+import com.azure.cosmos.rx.pojos.Pet;
+import com.azure.cosmos.util.CosmosPagedFlux;
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import org.testng.annotations.AfterClass;
+import org.testng.annotations.BeforeClass;
+import org.testng.annotations.Factory;
+import org.testng.annotations.Test;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+import java.util.Random;
+import java.util.UUID;
+import java.util.stream.Collectors;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+public class GroupByQueryTests extends TestSuiteBase {
+    List<Person> personList;
+    private CosmosAsyncContainer createdCollection;
+    private ArrayList<CosmosItemProperties> docs = new ArrayList<>();
+    private CosmosAsyncClient client;
+
+    @Factory(dataProvider = "clientBuildersWithDirect")
+    public GroupByQueryTests(CosmosClientBuilder clientBuilder) {
+        super(clientBuilder);
+    }
+
+    private static String getRandomName(Random rand) {
+        StringBuilder stringBuilder = new StringBuilder();
+        stringBuilder.append("name_" + rand.nextInt(100));
+
+        return stringBuilder.toString();
+    }
+
+    private static City getRandomCity(Random rand) {
+        int index = rand.nextInt(3);
+        switch (index) {
+            case 0:
+                return City.LOS_ANGELES;
+            case 1:
+                return City.NEW_YORK;
+            case 2:
+                return City.SEATTLE;
+        }
+
+        return City.LOS_ANGELES;
+    }
+
+    private static double getRandomIncome(Random rand) {
+        return rand.nextDouble() * Double.MAX_VALUE;
+    }
+
+    private static int getRandomAge(Random rand) {
+        return rand.nextInt(100);
+    }
+
+    @Test(groups = {"simple"}, timeOut = TIMEOUT)
+    public void queryDocuments() {
+        boolean qmEnabled = true;
+
+        String query = "SELECT sum(c.age) as sum_age, c.city FROM c group by c.city";
```

Comment:
There are a lot of other cases that need to be covered.

Reply:
Yup, need to add more tests for groupby. Plan is to get this in and run the Queryoracle. In the mean time will also try to add more tests